docs/nomad_usage.md

```diff
@@ -27,6 +27,41 @@ If a user requests a new runner, Poseidon duplicates the template Job of the cor
 
 When a user then executes their code, Poseidon copies the code into the container and executes it.
 
+### Nomad Restarts
+
+When the Nomad Servers or Agents restart, running executions can be terminated.  
+For agents, it depends on whether the runner allocation is placed on the restarted agent.
+For servers, it depends both on the role - if the restarted server is the cluster leader - and on whether Poseidon is connected to the restarted server, e.g. due to DNS Resolving.
+Poseidon can be connected to the server either for individual execution or for receiving the Nomad Event stream.
+The following table lists the behavior for restarts of Nomad Servers depending on its role and on whether Poseidon is connected to it (via DNS Resolution).
+
+| Role     | DNS Resolves | | WebSocket Problem? | Event Stream Problem? | 
+|----------|--------------|-|--------------------|-----------------------|
+| Leader   | Yes          | | problematic        | problematic           |
+| Leader   | No           | | problematic        | fine                  |
+| Follower | Yes          | | problematic        | problematic           |
+| Follower | No           | | fine               | fine                  |
+
+Such restarts lead to problems with either individual WebSocket connections of executions or the Nomad Event Stream.
+When the Nomad Event Stream is aborted, Poseidon tries to reestablish it. Once it succeeds in doing so, all environments and runners are recovered from Nomad.
+
+In the case of Nomad Agent restarts the WebSocket connection of a running execution aborts.
+Furthermore, when also Docker of the Nomad Agent is restarted, the containers are recreated.
+Poseidon captures such occurrences and uses the runner as clean and idle.  
+The Nomad and Docker systemd services are connected via a systemd PartOf relationship.
```

Comment:
Sure, but this is custom and not yet present in the docs, is it?

Reply:
I intended to introduce it with those two sentences 😅  Would you like to have a dedicated section in the docs?

Reply:
I thought it might make sense to add a sample file `/etc/systemd/system/nomad.service.d/override.conf` to the repo (or inline in the docs) to have the explicit settings. Otherwise, one needs to figure out which service depends on which over and over again (see the uncertainty we just had in the other thread).